chain/src/pipe.rs

```diff
@@ -108,7 +108,49 @@ pub fn process_block(b: &Block, mut ctx: BlockContext) -> Result<Option<Tip>, Er
 			extension.force_rollback();
 		}
 		Ok(h)
-	})
+	});
+
+	match result {
+		Ok(t) => {
+			sync_file_metadata(&b, &sumtrees, ctx.store.clone())?; 
+			Ok(t)
+		},
+		Err(e) => Err(e),
+	}
+}
+
+fn sync_file_metadata(b: &Block, sumtrees: &sumtree::SumTrees, store: Arc<ChainStore>) -> Result<(), Error> {
+	// We want to store written file data for the last 2 blocks, so we potentially
+	// have a couple of options to roll back to
+
+	// TODO: Delete all old blocks before doing this
+
+	// Save pmmr file metadata for this block
+	let block_file_md = sumtrees.last_file_metadata();
+	store
+		.save_block_pmmr_file_metadata(&b.hash(), &block_file_md)
+		.map_err(|e| Error::StoreErr(e, "saving pmmr file metadata".to_owned()))?;
+
+	// Delete the metadata of the oldest block written
+	let oldest_block_hash = store
+		.get_previous_pmmr_file_block();
+	if let Ok(h) = oldest_block_hash {
+		store
+			.delete_block_pmmr_file_metadata(&h)
+			.map_err(|e| Error::StoreErr(e, "deleting oldest stored pmmr block".to_owned()))?;
+	}
+
+	// Swap previous block index
```

Comment:
Instead of the 2 below wouldn't it be simpler (and more atomic) to save that directly in the chain head?

Reply:
Absolutely, good point, I can just add a field to the Tip struct and save it then, then we just have chain head and previous block position to deal with for the time being.